Add 2201.8.3 release note (#8294)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.8.3/RELEASE_NOTE.md

```
**Old Generated client resource/remote function**
```ballerina
    resource isolated function delete user/draft(string? userId = ()) returns http:Response|error {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        map<anydata> queryParam = {"userId": userId};
        resourcePath = resourcePath + check getPathForQueryParam(queryParam);
        http:Response response = check self.clientEp-> delete(resourcePath);
        return response;
    }
```
**New Generated client resource/remote function**
```ballerina
    resource isolated function delete user/draft(string? userId = ()) returns error? {
        string resourcePath = string `/user/draft`;
        map<anydata> queryParam = {"userId": userId};
        resourcePath = resourcePath + check getPathForQueryParam(queryParam);
        return self.clientEp->delete(resourcePath);
    }
```
### Bug fixes

To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)

## Developer tools updates

> ### Observability Improvements

Reviewer: ```suggestion
### Observability Improvements
```
Need to change the heading levels.

Reviewer 2: Addressed the suggestions

---
layout: ballerina-left-nav-release-notes
title: Swan Lake Update 8 (2201.8.3) 
permalink: /downloads/swan-lake-release-notes/2201.8.3/
active: 2201.8.3
---

## Overview of Ballerina Swan Lake Update 8 (2201.8.3)

<em>Swan Lake Update 8 (2201.8.3) is the third patch release of Ballerina 2201.8.0 (Swan Lake Update 8) and it includes a new set of bug fixes to the runtime and developer tooling.</em>

## Update Ballerina

Run the command below to update your current Ballerina installation directly to 2201.8.3 by using the [Ballerina Update Tool](/learn/update-tool/) as follows.

```
$ bal dist pull 2201.8.3
```

## Install Ballerina

If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.

## Runtime updates

### Bug fixes

To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).

> ## Ballerina library updates

Reviewer: Fixed.

## Developer tools updates

### Improvements

#### Observability

##### Provide `response_errors_total_value` metric

Ballerina observability now provides the total number of errors in responses as a metric via `response_errors_total_value`.

##### Define `service name` in traces

You can now add a suffix to the service name shown in the tracer providers ([Jaeger](https://www.jaegertracing.io/), [Zipkin](https://zipkin.io/), and [New Relic](https://newrelic.com/welcome-back)) by passing an environment variable in the runtime as given below. 

```
$ BAL_OBSERVE_SERVICE_NAME_SUFFIX=<suffix> bal run
```

> By default, the service name will be the base path of the service or the display name of the service.

Reviewer: Fixed.

### Bug fixes
To view bug fixes, see the GitHub milestone for Swan Lake Update 8 (2201.8.3) of the repositories below.

> - [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)

Author: open API one is a improvement, so it will not include in bugs

> - [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)

Reviewer: Fixed.

- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)

## Backward-incompatible changes

- OpenAPI tool client code generation has transitioned from representing the OpenAPI Specification byte format of the string type as Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.

> **OpenAPI sample**

Reviewer: ```suggestion
**OpenAPI sample**
```
These should be indented.

Reviewer: Fixed.

---
layout: ballerina-left-nav-release-notes
title: Swan Lake Update 8 (2201.8.3) 
permalink: /downloads/swan-lake-release-notes/2201.8.3/
active: 2201.8.3
---

## Overview of Ballerina Swan Lake Update 8 (2201.8.3)

<em>Swan Lake Update 8 (2201.8.3) is the third patch release of Ballerina 2201.8.0 (Swan Lake Update 8) and it includes a new set of bug fixes to the runtime and developer tooling.</em>

## Update Ballerina

Run the command below to update your current Ballerina installation directly to 2201.8.3 by using the [Ballerina Update Tool](/learn/update-tool/) as follows.

```
$ bal dist pull 2201.8.3
```

## Install Ballerina

If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.

## Runtime updates

### Bug fixes

To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).

> ## Library updates

Reviewer: ```suggestion
## Library updates
```
need to change the headings.

#### OpenAPI Tool

> - Code generation has transitioned from representing the OpenAPI Specification byte format of string type as the Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.

Reviewer: ```suggestion
- Code generation has transitioned from representing the OpenAPI Specification byte format of the string type as Ballerina `byte[]` to using Ballerina `string` type due to an issue with the previously generated code. This change will break the compatibility with the existing generated code.
```

> **OpenAPI Sample**

Reviewer: ```suggestion
**OpenAPI sample**
```

```yaml
components:
 schemas:
  ByteExample:
   type: string
   format: byte
```

> **Old Generated Ballerina code**

Reviewer: ```suggestion
**Old generated Ballerina code**
```

```ballerina
type ByteExample byte[];
```

> **New Generated Ballerina code**

Reviewer: ```suggestion
**New generated Ballerina code**
```

```ballerina
type ByteExample string;
```

> - Changed the client resource/remote code generation of `http:Response` to a `nil` return type when the operation in the OpenAPI specification has a response with no content. With this improvement, the already generated code will be altered when regenerated with version the Ballerina 2201.8.3.

Reviewer: ```suggestion
- Changed the `client resource/remote code` generation of the `http:Response` to a `nil` return type when the operation in the OpenAPI specification has a response with no content. With this improvement, the already generated code will be altered when regenerated with the Ballerina Swan Lake Update 8 (2201.8.3) version.
```

> **OpenAPI Sample**

Reviewer: ```suggestion
**OpenAPI sample**
```

```yaml
 /user/draft:
    delete:
      operationId: deleteDraft
      parameters:
        - in: query
          name: userId
          schema:
            type: string    
      responses:
        "200":
          description: Successful response //No content type
```

> **Old Generated client resource/remote function**

Reviewer: ```suggestion
**Old generated client resource/remote function**
```

```ballerina
    resource isolated function delete user/draft(string? userId = ()) returns http:Response|error {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        map<anydata> queryParam = {"userId": userId};
        resourcePath = resourcePath + check getPathForQueryParam(queryParam);
        http:Response response = check self.clientEp-> delete(resourcePath);
        return response;
    }
```

> **New Generated client resource/remote function**

Reviewer: ```suggestion
**New generated client resource/remote function**
```

```ballerina
    resource isolated function delete user/draft(string? userId = ()) returns error? {
        string resourcePath = string `/user/draft`;
        map<anydata> queryParam = {"userId": userId};
        resourcePath = resourcePath + check getPathForQueryParam(queryParam);
        return self.clientEp->delete(resourcePath);
    }
```
### Bug fixes

To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)

## Developer tools updates

### Improvements

#### Observability

> ##### Provide `response_errors_total_value` metric

Reviewer: ```suggestion
##### Provide the `response_errors_total_value` metric
```

> Ballerina observability now provides total number of errors in responses as an metric via `response_errors_total_value`.

Reviewer: ```suggestion
Ballerina observability now provides the total number of errors in responses as a metric via `response_errors_total_value`.
```

> ##### Let the user define `service name` in traces

Author: ```suggestion
##### Define `service name` in traces
```

> ##### Let the user define `service name` in traces

Reviewer: ```suggestion
##### Define the `service name` in traces
```

Reviewer 2: Addressed the suggestions

> User can now add a suffix to the service name shown in the tracer providers (Jaeger, Zipkin & New Relic) by passing an environment variable in the runtime as given below. 

Reviewer: ```suggestion
You can now add a suffix to the service name shown in the tracer providers (Jaeger, Zipkin, and New Relic) by passing an environment variable in the runtime as given below. 
```

---
layout: ballerina-left-nav-release-notes
title: Swan Lake Update 8 (2201.8.3) 
permalink: /downloads/swan-lake-release-notes/2201.8.3/
active: 2201.8.3
---

## Overview of Ballerina Swan Lake Update 8 (2201.8.3)

<em>Swan Lake Update 8 (2201.8.3) is the third patch release of Ballerina 2201.8.0 (Swan Lake Update 8) and it includes a new set of bug fixes to the runtime and developer tooling.</em>

## Update Ballerina

Run the command below to update your current Ballerina installation directly to 2201.8.3 by using the [Ballerina Update Tool](/learn/update-tool/) as follows.

```
$ bal dist pull 2201.8.3
```

## Install Ballerina

If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.

## Runtime updates

### Bug fixes

To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).

> 

Reviewer 3: ```suggestion

## Library updates
### Breaking changes

#### OpenAPI Tool
- Transitioned code generation from representing OpenAPI Specification byte format of string type as Ballerina `byte[]` to using Ballerina `string` type.

**OpenAPI Sample**
```Yaml
components:
  schemas:
       ByteExample:
	      type: string
              format: byte
```
**Old Generated Ballerina code**
```ballerina
type ByteExample byte[];
```
**New Generated Ballerina code**
```ballerina
type ByteExample string;
```
- Changed the client resource/remote code generation of `http:Response` to `nil` return type when the operation of OpenAPI specification has a response with no content.

**OpenAPI Sample**
```yaml
 /users/drafts/{id}:
    delete:
      operationId: deleteDraft
      parameters:
        - in: path
          name: id
          required: true
          schema:
            type: string
      responses:
        "200":
          description: Successful response //No content type
```
**Old Generated client resource/remote function**
```ballerina
    resource isolated function delete users/drafts/[string id]() returns http:Response|error {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        http:Response response = check self.clientEp-> delete(resourcePath);
        return response;
    }
```
**New Generated client resource/remote function**
```ballerina
resource isolated function delete users/drafts/[string id]() returns error? {
        string resourcePath = string `/users/drafts/${getEncodedUri(id)}`;
        return self.clientEp-> delete(resourcePath);
}
```
### Bug fixes

To view bug fixes, see the GitHub milestone for Swan Lake Update 2201.8.3 of the repositories below.
- [OpenAPI](https://github.com/ballerina-platform/ballerina-library/issues?q=is%3Aissue+label%3Amodule%2Fopenapi-tools+label%3AType%2FBug+milestone%3A2201.8.3+is%3Aclosed)
```

> 

Reviewer: Reviewed.

---
layout: ballerina-left-nav-release-notes
title: Swan Lake Update 8 (2201.8.3) 
permalink: /downloads/swan-lake-release-notes/2201.8.3/
active: 2201.8.3

> redirect_from: 

Reviewer: ```suggestion
redirect_from: 
```
Need to remove these redirections as they are not used anymore. Please check the template.

    - /downloads/swan-lake-release-notes/2201.8.3
    - /downloads/swan-lake-release-notes/2201.8.3/
    - /downloads/swan-lake-release-notes/2201.8.3-swan-lake/
    - /downloads/swan-lake-release-notes/2201.8.3-swan-lake
---

## Overview of Ballerina Swan Lake Update 8 (2201.8.3)

<em>Swan Lake Update 8 (2201.8.3) is the third patch release of Ballerina 2201.8.0 (Swan Lake Update 8) and it includes a new set of bug fixes to the runtime and developer tooling.</em>

## Update Ballerina

Run the command below to update your current Ballerina installation directly to 2201.8.3 by using the [Ballerina Update Tool](/learn/update-tool/) as follows.

```
$ bal dist pull 2201.8.3
```

## Install Ballerina

If you have not installed Ballerina, download the [installers](/downloads/#swanlake) to install.

## Runtime updates

### Bug fixes

> To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).

Reviewer: ```suggestion
To view bug fixes, see the [GitHub milestone for 2201.8.3 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FjBallerina).
```
This milestone link is empty.

Author: I think there is no difference with suggested changes

## Developer tools updates

### Bug fixes
To view bug fixes, see the GitHub milestone for Swan Lake Update 8 (2201.8.3) of the repositories below.

> - [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)

Reviewer: ```suggestion
- [Dev Tools](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.8.3+label%3ATeam%2FDevTools+label%3AType%2FBug+is%3Aclosed)
```
This is also empty.

Author: I think there is no difference with suggested changes

Reviewer: The link is empty.

Author: It is because the issues are not closed since master PRs are not merged

Author: Is it affect to the release note?

Reviewer: We can't have empty links on the release note. The release note should be released together with the release, which means, after completing the fixes and closing the issues.

Author: We have to create a separate issues for 8.x then
I will work on that
